phoenix-core/src/main/java/org/apache/phoenix/schema/PMetaDataImpl.java

```diff
@@ -160,7 +160,7 @@ public PMetaDataCache cloneMinusOverage(long overage) {
                     newCache.put(tableRef.table.getKey(), new PTableRef(tableRef));
                     toRemove.add(tableRef);
                     toRemoveBytes += tableRef.estSize;
-                    if (toRemoveBytes - toRemove.peekLast().estSize > overage) {
+                    while (toRemoveBytes - toRemove.peekLast().estSize >= overage) {
```

Comment:
Probably worth a test or two in PMetaDataImplTest. I haven't worked with this code in a while, but I seem to remember that we keep at least one item in PMetaDataImpl, even if it's larger than the configured max size (which would imply that we should break out of the loop when the size is one). Also, what if the max size is configured to be zero?